File: Backend/src/main/java/com/copernic/backend/Backend/entity/PuntBescanvi.java
package com.copernic.backend.Backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

import java.util.List;

/**
 * Entitat que representa un punt de bescanvi dins del sistema.
 * Un punt de bescanvi és un lloc físic on els usuaris poden recollir recompenses.
 * Conté informació de contacte, localització i les recompenses associades.
 */
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@ToString
public class PuntBescanvi {

    /**
     * Identificador únic del punt de bescanvi.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * Nom del punt de bescanvi.
     */
    @Column
    private String nom;

    /**
     * Codi postal on es troba el punt de bescanvi.
     */
    @Column
    private String codiPostal;

    /**
     * Adreça física del punt de bescanvi.
     */
    @Column
    private String adreca;

    /**
     * Telèfon de contacte del punt de bescanvi.
     */
    @Column
    private String telefon;

    /**
     * Nom de la persona de contacte responsable del punt.
     */
    @Column
    private String personaContacte;

    /**
     * Correu electrònic de contacte del punt de bescanvi.
     */
    @Column
    private String email;

    /**
     * Observacions o comentaris addicionals sobre el punt de bescanvi.
     */
    @Column
    private String observacions;

    /**
     * Imatge representativa del punt de bescanvi, emmagatzemada com a cadena en Base64.
     */
    @Lob
    private String foto;

    /**
     * Llista de recompenses associades a aquest punt de bescanvi.
     * Relació OneToMany amb la classe {@link Recompensas}.
     * S'exclou del {@code toString} i de la serialització JSON per evitar referències circulars i sobrecàrrega.
     */
    @OneToMany(mappedBy = "puntBescanviId")
    @ToString.Exclude
    @JsonIgnore
    private List<Recompensas> puntBescanviID;
}

File: Frontend/app/src/main/java/cat/copernic/amayo/frontend/puntBesacanviManagment/model/PuntBescanvi.kt
package cat.copernic.amayo.frontend.puntBesacanviManagment.model

/**
 * Representa un punt de bescanvi dins del sistema.
 *
 * Aquesta classe conté la informació associada a un punt físic on es poden bescanviar recompenses.
 * Inclou dades de contacte, localització i una imatge representativa en format Base64.
 *
 * @property id Identificador únic del punt de bescanvi.
 * @property nom Nom del punt de bescanvi.
 * @property codiPostal Codi postal de la ubicació.
 * @property adreca Adreça física del punt de bescanvi.
 * @property telefon Telèfon de contacte del punt.
 * @property personaContacte Persona de referència per a contactes.
 * @property email Correu electrònic de contacte.
 * @property observacions Notes o observacions addicionals.
 * @property foto Imatge del punt en format Base64 (opcional).
 */
data class PuntBescanvi (
    var id: Long? = null,
    var nom: String? = null,
    var codiPostal: String? = null,
    var adreca: String? = null,
    var telefon: String? = null,
    var personaContacte: String? = null,
    var email:String? = null,
    var observacions: String? = null,
    var foto: String? = null
)